providers/dns/azure/azure.go

```diff
@@ -169,9 +170,15 @@ func (d *DNSProvider) Present(domain, token, keyAuth string) error {
 	ctx := context.Background()
 	fqdn, value := dns01.GetRecord(domain, keyAuth)
 
-	zone, err := d.getHostedZoneID(ctx, fqdn)
-	if err != nil {
-		return fmt.Errorf("azure: %w", err)
+	var zone string
+	if env.GetOrFile(EnvZoneName) != "" {
+		zone = env.GetOrFile(EnvZoneName)
+	} else {
+		var err error
+		zone, err = d.getHostedZoneID(ctx, fqdn)
+		if err != nil {
+			return fmt.Errorf("azure: %w", err)
+		}
```

Comment:
This should be moved into `getHostedZoneID`.

Reply:
Agree, done so.